prometheusbp/promtest/testutil.go

```diff
@@ -10,24 +10,36 @@ import (
 
 // PrometheusMetricTest stores information about a metric to use for testing.
 type PrometheusMetricTest struct {
-	tb        testing.TB
-	metric    prometheus.Collector
-	name      string
-	initValue float64
-	labels    prometheus.Labels
+	tb             testing.TB
+	metric         prometheus.Collector
+	name           string
+	initValue      float64
+	initHistoCount int
+	labels         prometheus.Labels
 }
 
 // CheckDelta checks that the metric value changes exactly delta from when Helper was called.
 func (p *PrometheusMetricTest) CheckDelta(delta float64) {
 	p.tb.Helper()
 
-	got := p.getValue()
+	got, _ := p.getValues()
 	got -= float64(p.initValue)
 	if got != delta {
 		p.tb.Errorf("%s metric delta: wanted %v, got %v", p.name, delta, got)
 	}
 }
 
+// CheckHistoCountDelta checks that the number of samples of histogram is the exactly delta from when Helper was called.
+func (p *PrometheusMetricTest) CheckHistoCountDelta(delta int) {
```

Comment:
I don't like abbreviating words in exported API names.

CheckSampleDelta?

Reply:
I actually think the current implementation of `CheckDelta` for histogram is a mistake, what does a delta value for a histogram mean, exactly? If we just change histogram to store the sample count in its `initValue` and change its `CheckDelta` to check the sample count delta, that addresses both of your concerns, what do you think?

technically that is a breaking change (if someone has some code actually using `CheckDelta` on histograms it will break), but I really don't know how that is usable right now.

cc @JessicaGreben

Reply:
oh actually someone are actually using it to check the observed latency seconds. ok I'll just spell `Histogram` fully instead. I don't think the name `CheckSampleDelta` emphasize enough that it's for histograms only.

Reply:
Both histograms and summaries have samples, so it makes sense for more than just histograms.

Reply:
oh shit I didn't see this comment before merging it. will revert and redo it.